frontends/bmd/src/app_polls_flows.test.tsx

```diff
@@ -0,0 +1,1352 @@
+    precinctSelection: singlePrecinctSelectionFor('23'),
+    isLiveMode: false,
+    pollsTransition: 'close_polls',
+    ballotCounts: {
+      'undefined,__ALL_PRECINCTS': [20, 5],
+      'undefined,23': [20, 5],
+    },
+  };
+
+  card.insertCard(pollWorkerCard, JSON.stringify(pollsClosedCardTallyReport));
+  await screen.findByText('Polls Closed Report on Card');
+  screen.getByText(/contains a polls closed report/);
+  screen.getByText(/the polls will be closed on VxMark/);
+  userEvent.click(screen.getByText('Close Polls and Print Report'));
+  await screen.findByText('Printing polls closed report');
+  function checkPollsClosedReport(printedElement: RenderResult) {
+    const generalReport = printedElement.getByTestId(
+      'tally-report-undefined-23'
+    );
+    within(generalReport).getByText(
+      'TEST Polls Closed Report for Center Springfield'
+    );
+    expectBallotCountsInReport(generalReport, 20, 5, 25);
+    expectContestResultsInReport(
+      generalReport,
+      'president',
+      {
+        ballotsCast: 34,
+        undervotes: 6,
+        overvotes: 0,
+      },
+      {
+        'barchi-hallaren': 6,
+        'cramer-vuocolo': 5,
+        'court-blumhardt': 6,
+        'boone-lian': 5,
+        'hildebrand-garritty': 3,
+        'patterson-lariviere': 0,
+      }
+    );
+    expectEmptyContestResultsInReport(generalReport, 'senator');
+    expectEmptyContestResultsInReport(generalReport, 'governor');
+  }
+  await expectPrint(checkPollsClosedReport);
+  jest.advanceTimersByTime(REPORT_PRINTING_TIMEOUT_SECONDS * 1000);
+  await screen.findByText('Polls Closed Report Printed');
+  expect(card.writeLongUint8Array).toHaveBeenCalledTimes(4);
+  expect(card.writeLongUint8Array).toHaveBeenCalledWith(
+    expect.objectContaining([])
+  ); // Card cleared
+  expect(logger.log).toHaveBeenCalledWith(
+    LogEventId.PollsClosed,
+    'poll_worker',
+    expect.anything()
+  );
+  expect(logger.log).toHaveBeenCalledWith(
+    LogEventId.TallyReportPrinted,
+    'poll_worker',
+    expect.objectContaining({
+      message:
+        'Printed 2 copies of a polls closed report for Center Springfield exported from scanner 001.',
+    })
+  );
+  expect(logger.log).toHaveBeenCalledWith(
+    LogEventId.TallyReportClearedFromCard,
+    'poll_worker',
+    expect.anything()
+  );
+  userEvent.click(screen.getByText('Print Additional Report'));
+  await screen.findByText('Printing polls closed report');
+  await expectPrint(checkPollsClosedReport);
+  jest.advanceTimersByTime(REPORT_PRINTING_TIMEOUT_SECONDS * 1000);
+  await screen.findByText('Polls Closed Report Printed');
+  userEvent.click(screen.getByText('Continue'));
+  screen.getByText(hasTextAcrossElements('Polls: Closed'));
+  card.removeCard();
+  await screen.findByText('Polls Closed');
+  screen.getByText('Voting is complete.');
+});
+
+test('tally report: as expected with all precinct combined data for general election', async () => {
```

Comment:
Moved tests from `poll_worker_screen.test.tsx` to this new file because I wanted all the polls flows / report assertions in one file, so all these `tally report: as expected` tests are just copied over and edited.